midp/localmsg.js

```diff
@@ -1041,6 +1057,11 @@ Native["org/mozilla/io/LocalMsgConnection.init.(Ljava/lang/String;)V"] = functio
 
     this.connection = typeof MIDP.LocalMsgConnections[this.protocolName] === 'function' ?
         new MIDP.LocalMsgConnections[this.protocolName]() : MIDP.LocalMsgConnections[this.protocolName];
+
+    var ctx = $.ctx;
+    this.connection.reset();
+    ctx.setAsCurrentContext();
```

Comment:
Maybe the context switching should happen as part of the call to `reset` so that callers don't have to remember to do that?